MaxText/checkpointing.py

```diff
@@ -178,8 +178,11 @@ def  map_to_pspec(data):
     p = epath.Path(load_parameters_from_path)
     ckptr = orbax.checkpoint.PyTreeCheckpointer()
     restore_args = orbax.checkpoint.checkpoint_utils.construct_restore_args(abstract_unboxed_pre_state.params)
+    # Orbax quirk -> we save the entire TrainState, which has a field `params` which holds the PyTree
```

Comment:
Hmmm this seems weird, should we fix that while we're fixing things?

Reply:
Ack, I'll double check with Susie who worked on this last I think. If I had to answer now, I think just saving the state directly is useful because we also get the optimizer state captured in the checkpoint. And we can cleanly capture any other variable collections (like `overwrite_with_gradients`) without any sort of special logic.

Reply:
Update #2: We can actually just tweak this a little bit to just load things used the abstract_state directly and it works. Cleaner code, but I wonder if this is less efficient/wasteful because we are restoring things we don't need (i.e. all the stuff that isn't `params`)

Reply:
And another update: This actually seems to be an optimization. We don't want to restore anything other than the `params` field here as that would be wasteful, both computationally and for memory usage (we immediately toss the rest of it). So we have this little odd bit here to basically say only restore the params field. I added a comment clarifying what's really happening.